uvicorn/protocols/http/httptools_impl.py

```diff
@@ -190,6 +171,29 @@ def handle_upgrade(self):
         protocol.data_received(b"".join(output))
         self.transport.set_protocol(protocol)
 
+    def send_400_response(self, msg: str):
+
+        from uvicorn.protocols.websockets.auto import AutoWebSocketsProtocol
+
+        if AutoWebSocketsProtocol is None:
+            msg = "No supported WebSocket library detected. Please use 'pip install uvicorn[standard]', or install 'websockets' or 'wsproto' manually."  # noqa: E501
+            self.logger.warning(msg)
```

Comment:
This looks like it needs to be deleted.

Reply:
this was part of improving user experience in https://github.com/encode/uvicorn/pull/926 @tomchristie 
Do you want we get rid of it ?

Reply:
Perhaps it just needs to stay within the `if upgrade_value != b"websocket" or self.ws_protocol_class is None:` block. It doesn't make sense for this to be within `send_400_response` right? (Eg. it shouldn't log if it's called because of `except httptools.HttpParserError as exc:`)

Reply:
ha yes you're right I didn't notice if was sent also in that case, I misread thinking it was just for the upgrade case